STATE-COMPATIBILITY.md

```diff
@@ -0,0 +1,212 @@
+# State-Compatibility
+
+- [State-Compatibility](#state-compatibility)
+  - [Scope](#scope)
+  - [Validating State-Compatibility](#validating-state-compatibility)
+    - [AppHash](#apphash)
+    - [LastResultsHash](#lastresultshash)
+  - [Major Sources of State-incompatibility](#major-sources-of-state-incompatibility)
+    - [Creating Additional State](#creating-additional-state)
+    - [Changing Proto Field Definitions](#changing-proto-field-definitions)
+    - [Returning Different Errors Given Same Input](#returning-different-errors-given-same-input)
+    - [Variability in Gas Usage](#variability-in-gas-usage)
+  - [Secondary Limitations To Keep In Mind](#secondary-limitations-to-keep-in-mind)
+    - [Network Requests to External Services](#network-requests-to-external-services)
+    - [Randomness](#randomness)
+    - [Parallelism and Shared State](#parallelism-and-shared-state)
+    - [Hardware Errors](#hardware-errors)
+
+
+It is critical for the patch releases to be state-machine compatible with prior releases in the same minor version. 
+For example, v3.2.1 must be state-machine compatible with v3.2.0. 
+
+This is to ensure **determinism**, i.e., given the same input, the nodes will always produce the same output.
+
+State-incompatibility is allowed for major upgrades because all nodes in either the consumer or provider networks
+perform it at the same time. Therefore, after the upgrade, the nodes continue functioning in a deterministic way.
+ 
+## Scope
+
+The state-machine scope includes the following areas:
+
+- All ICS messages including:
+  - Every msg's ValidateBasic method
+  - Every msg's MsgServer method
+    - Net gas usage, in all execution paths
+    - Error result returned
+    - State changes (namely every store write)
+- AnteHandlers in "DeliverTx" mode
+- All `BeginBlock`/`EndBlock` logic
+
+The following are **NOT** in the state-machine scope:
+
+- Events
```

Comment:
Not sure about events being outside the scope, is there a source claiming this?

Reply:
CometBFT needs events to be the same. See here:
https://github.com/cometbft/cometbft/blob/main/spec/abci/abci%2B%2B_basic_concepts.md#deterministic-state-machine-replication

Quite

> Note that some methods (Query, FinalizeBlock) return non-deterministic data in the form of Info and Log fields. The Log is intended for the literal output from the Application's logger, while the Info is any additional info that should be returned. These are the only fields that are not included in block header computations, so we don't need agreement on them. All other fields in the Response* must be strictly deterministic.

Emphasis on this:
> All other fields in the Response* must be strictly deterministic.

Reply:
Based on feedback from the Comet team, events are indeed out of the state-machine scope.